lib/util.js

```diff
@@ -128,6 +128,28 @@ exports.chooseFormat = (formats, options) => {
       format = formats[formats.length - 1];
       break;
 
+    case 'highestaudio':
+      var highest = formats[0];
+      for (var format of formats) {
+        if (!highest.audioBitrate || (format.audioBitrate
+          && (format.audioBitrate > highest.audioBitrate
+          || (format.audioBitrate === highest.audioBitrate && !format.encoding))))
+            highest = format;
+      }
+      format = highest;
+      break;
+
+    case 'highestvideo':
+      var highest = formats[0];
+      for (var format of formats) {
+        if (!highest.bitrate || (format.bitrate
+          && (Number(format.bitrate.split('-')[0]) > Number(highest.bitrate.split('-')[0])
+          || (Number(format.bitrate.split('-')[0]) === Number(highest.bitrate.split('-')[0]) && !format.audioEncoding))))
```

Comment:
Same comments as above apply to this `if` statement. Also, the bitrate can be a float, and `util.sortFormats` looks at the right side of `bitrate`

https://github.com/fent/node-ytdl-core/blob/master/lib/util.js#L63